### PR TITLE
GH-3189 dismiss screen explicitly

### PR DESCRIPTION
**Pull request by biafra23**
Handles #3189

Changes proposed in this pull request:
- dismiss controller
-
-

### CODE REVIEW

File: Multisig/UI/Safe Management/Create Safe/SafeCreatingViewController.swift [hunk 1]
```diff
@@ -75,6 +75,7 @@ class SafeCreatingViewController: UIViewController {
         Timer.scheduledTimer(withTimeInterval: 3, repeats: false) { [weak self] timer in
             DispatchQueue.main.async {
                 self?.onSuccess()
+                self?.dismiss(animated: true)
```
Comment (biafra23) [inline Multisig/UI/Safe Management/Create Safe/SafeCreatingViewController.swift]: This fixes the bug, but there might be a better way

Comment (DmitryBespalov) [reply Multisig/UI/Safe Management/Create Safe/SafeCreatingViewController.swift]: I think this is misplaced: we put the dismiss of a view controller near the place where it was opened (on the calling side)

Comment (MouazAlzahabi) [reply Multisig/UI/Safe Management/Create Safe/SafeCreatingViewController.swift]: same here, specially that we have onSuccess handler

File: Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift [hunk 2]
```diff
@@ -113,6 +113,7 @@ class CreateSafeFlow: UIFlow, ASAuthorizationControllerPresentationContextProvid
             let view = self.factory.safeCreatingViewController()
             view.onSuccess = { [weak self] in
```
Comment (MouazAlzahabi) [inline Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift]: I think this is wrong, 

As it's presented using "show" function then is should be dismissed when the flow stop function and dismiss should have no effect on it,

Comment (DmitryBespalov) [reply Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift]: second that

Comment (biafra23) [reply Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift]: The presenter (LoadSafeViewController) is dismissed in UIFlow.stop() and the Flow is set to nil in the completion closure, but (without this change) the view is not dismissed.

Comment (MouazAlzahabi) [reply Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift]: Then maybe we have a referencing problem and the correct fix to solve it

**Review by DmitryBespalov — APPROVED** [review]

File: Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift [hunk 3]
```diff
@@ -89,10 +89,14 @@ class CreateSafeFlow: UIFlow, ASAuthorizationControllerPresentationContextProvid
...
         appleWeb3AuthLogin = AppleWeb3AuthLogin(
-            authorizationComplete: {
+            authorizationComplete: { [weak self] in
+                guard let self = self else { return }
                 let view = self.factory.safeCreatingViewController()
-                view.onSuccess = {
-                    self.stop(success: true)
+                view.onSuccess = { [weak self, unowned view] in
+                    view.dismiss(animated: true) {
+                        // The dismiss() should not be necessary here, biut the SafeCreatingViewCobntroller is not dismissed, if omitted
```
Comment (DmitryBespalov) [inline Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift]: typo

**Pull request merged**